src/util.rs

```diff
@@ -275,17 +287,34 @@ pub fn run_hook(
     let hook_path = repo
         .config()?
         .get_path("core.hooksPath")
-        .unwrap_or_else(|_| repo.path().join("hooks"));
-    if hook_path.join(hook_name).exists() {
+        .unwrap_or_else(|_| repo.path().join("hooks"))
+        .join(hook_name);
+    if hook_path.exists() {
+        let GitRunInfo {
+            // We're calling a Git hook, but not Git itself.
+            path_to_git: _,
+            // We always want to call the hook in the Git working directory.
+            working_directory: _,
+            env,
+        } = git_run_info;
         let mut child = Command::new(get_sh().context("shell needed to run hook")?)
             .arg("-c")
-            // This is always passed as a current dir unix path, as `sh` is processing it as an
-            // argument, even on Windows.
-            .arg(format!("./{} \"$@\"", hook_name))
+            .arg(format!(
+                "{} \"$@\"",
```

Comment:
FYI this argument needed the `./` prefix on Windows it wouldn't look in current directory otherwise.  I think this is due to gitbash's weird argument path translation, but am not sure.

Reply:
Technically, there's no stipulation that the hook directory has to be somewhere under the current repository, such as if `git.core.hooksPath` is set to something else, so I think it's wrong to use `./` in this case.

Reply:
Hmm actually I don't understand why the `./` isn't needed in Linux also: it isn't like we are expecting `.` to be in $PATH(?)

Reply:
Ah I see you removed the `.current_dir(hook_path)`  which made the `./` correct 😄

Reply:
I see what is happening.  See this line in the `bash` manpage:

```
If  the name is neither a shell function nor a builtin, and contains no slashes, bash searches each element of the PATH for a directory containing an executable file by that name.
```

and later

```If the search is successful, or if the command name contains one or more slashes, the shell executes the named program in a separate execution  environment.
```

Because `std::fs::canonicalize(&hook_path)` most certainly does contain `/`, the shell is interpreting it as a path to binary, whereas on Windows, lacking a '/' it is falling back to searching via PATH.  This explains why it worked for me when I added the `./` prefix unconditionally.


`std::fs::canonicalize(&hook_path)`

Reply:
Thinking about ways this could be fixed without resorting to path mangling, it feels like the simplest way to resolve is to just add the canonicalized hook_path to the start of PATH and invoke the hook by name directly.